### PR TITLE
Handle exception when refresh table since table may be dropped or re-created

### LINKED ISSUE #3076: Query error  unstable from iceberg external table when iceberg table dropped
### Steps to reproduce the behavior (Required)
1. create iceberg table:
```
create table hive_prod.iceberg_db.iceberg_par_parquet_snappy_refresh(
col_boolean boolean,
col_int int,
col_long long,
col_float float,
col_double double,
col_decimal decimal(38,18),
col_date date,
col_timestamp timestamp,
col_string string,
col_binary binary,
col_struct struct<a:string,b:int>,
col_map map<string,int>
) USING iceberg 
PARTITIONED BY (col_date)
TBLPROPERTIES('write.format.default'='parquet','write.parquet.compression-codec'='snappy');
```

2. create iceberg external table:
```
create external table ex_iceberg_parquet_refresh (
col_int int,
col_date date,
col_string varchar(200),
col_boolean boolean,
col_float float,
col_double double,
col_long bigint,
col_decimal decimal(38,18))
ENGINE=iceberg
properties (
"resource" = "iceberg_emr_tn",
"table" = "iceberg_par_parquet_snappy_refresh",
"database" = "iceberg_db");
```

3. drop iceberg table:
```
drop table hive_prod.iceberg_db.iceberg_par_parquet_snappy_refresh;
```

4. select iceberg external table:
```
mysql> select * from ex_iceberg_parquet_refresh;
ERROR 1064 (HY000): No such table: iceberg_db.iceberg_par_parquet_snappy_refresh
mysql> select * from ex_iceberg_parquet_refresh;
ERROR 1064 (HY000): Unknown error
```

### Expected behavior (Required)
ERROR 1064 (HY000): No such table: iceberg_db.iceberg_par_parquet_snappy_refresh

### Real behavior (Required)
After the second query, the error becomes:
ERROR 1064 (HY000): Unknown error

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+------------------------+
| current_version()      |
+------------------------+
| QA_TEST_MASTER d4a0de6 |
+------------------------+
1 row in set (0.04 sec)
```

### LINKED ISSUE #3078: "Table UUID does not match" error needed after recreate iceberg table
### Steps to reproduce the behavior (Required)
1. create iceberg table;
2. create iceberg external table;
3. drop iceberg table --> create iceberg table with same table name --> select iceberg external table:
```
mysql> select * from ex_iceberg_parquet_refresh;
ERROR 1064 (HY000): Table UUID does not match: current=1435706f-1321-4c8f-ab4b-6b248ceaaae7 != refreshed=612411ac-4f8e-425f-a23a-cb76f79b242f
```
4. drop iceberg table --> select iceberg external table --> create iceberg table with same table name --> select iceberg external table, no error.

### Expected behavior (Required)
Table UUID does not match error reported after recreate iceberg table.

### Real behavior (Required)
some times no error reported after recreate iceberg table.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+------------------------+
| current_version()      |
+------------------------+
| QA_TEST_MASTER d4a0de6 |
+------------------------+
1 row in set (0.01 sec)
```

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [x] bug

## Which issues of this PR fixes ：

Fixes #3076 and Close #3078

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 1]
```diff
@@ -104,4 +106,23 @@ public static TableScan getTableScan(Table table,
...
+            table.refresh();
+            if (table instanceof BaseTable && ((BaseTable) table).operations().current() == null) {
+                // TableOperation like HiveTableOperations will not throw NoSuchTableException after the first time.
+                // We should throw here to let user get the same semantic for this case.
+                // See: https://github.com/StarRocks/starrocks/issues/3076
+                throw new NoSuchTableException("No such table: %s", table.name());
+            }
+        } catch (NoSuchTableException en) {
+            throw new StarRocksIcebergException("Refresh table with failure ", en);
+
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:121]: redundant empty line

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 2]
```diff
@@ -104,4 +106,24 @@ public static TableScan getTableScan(Table table,
...
+            table.refresh();
+            if (table instanceof BaseTable && ((BaseTable) table).operations().current() == null) {
+                // TableOperation like HiveTableOperations will not throw NoSuchTableException after the first time.
+                // We should throw here to let user get the same semantic for this case.
+                // See: https://github.com/StarRocks/starrocks/issues/3076
+                throw new NoSuchTableException("No such table: %s", table.name());
+            }
+        } catch (NoSuchTableException e) {
+            throw new StarRocksIcebergException(String.format("No such table  %s", table.name()));
+
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:121]: remove redundant empty line

**Review by DorianZheng — DISMISSED** [review]
rest LGTM

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 3]
```diff
@@ -104,4 +106,24 @@ public static TableScan getTableScan(Table table,
         }
         return tableScan;
     }
+
+    private static void refreshTable(Table table) {
+        try {
+            table.refresh();
+            if (table instanceof BaseTable && ((BaseTable) table).operations().current() == null) {
```
[the remark this answers is not in the document]
Comment (caneGuy) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java]: when we get refresh and get tablemetadata with null value, it indicates that table has been dropped

**Review by DorianZheng — DISMISSED** [review]
LGTM

**Review by HangyuanLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 4]
```diff
@@ -104,4 +106,28 @@ public static TableScan getTableScan(Table table,
...
+            } else {
+                // table loaded by Catalog should be a base table
+                throw new StarRocksIcebergException(String.format("Invalid table type of %s, it should be a BaseTable!",
+                        table.name()));
+            }
+        } catch (NoSuchTableException e) {
+            throw new StarRocksIcebergException(String.format("No such table  %s", table.name()));
+        } catch (IllegalStateException ei) {
+            throw new StarRocksIcebergException(String.format("Refresh table %s with failure, the table under hood" +
+                    " may have been dropped. You should re create the external table. cause %s",
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:129]: re create -> re-create ?

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 5]
```diff
@@ -104,4 +106,28 @@ public static TableScan getTableScan(Table table,
...
+
+    private static void refreshTable(Table table) {
+        try {
+            if (table instanceof BaseTable) {
+                BaseTable baseTable = (BaseTable) table;
+                if (baseTable.operations().refresh() == null) {
+                    // If table is loaded successfully, current table metadata will never be null.
+                    // So when we get a null metadata after refresh, it indicates the table has been dropped.
+                    // See: https://github.com/StarRocks/starrocks/issues/3076
+                    throw new NoSuchTableException("No such table: %s", table.name());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:118]: LGTM

**Review by HangyuanLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**